src/machine.js

```diff
@@ -87,6 +87,14 @@ class Machine {
     return s
   }
 
+  get_all_groups() {return Object.keys(this.get_data().groups || {'': null})}
+  get_groups_with_resources() {
+    let groups = []
+    for (let group of this.get_all_groups())
+      if (this.get_config(group).cpus || this.get_gpus(group).length >= 1) groups.push(group)
+    return groups
+  }
+
```

Comment:
I don't like splitting the groups as this could alter other behavior.  I'd rather detect which groups are empty in the view part of the code then hide them when ``config.hide_empty_groups`` is true.  This is more inline with the rest of the code and keeps a better separation between data and view.